### PR TITLE
[browser][binding]Fix automatic signature detection for executable entry point

### PR DESCRIPTION
**Pull request by kjpou1**
with string[] type for arguments.

### CODE REVIEW

File: src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs [hunk 1]
```diff
@@ -323,6 +323,10 @@ private static string GetCallSignature(IntPtr methodHandle)
                         {
                             res[c] = 'h';
                         }
+                        else if (t == typeof(string[]))
+                        {
+                            res[c] = 'm';
+                        }
```
Comment (radical) [inline src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs:329]: Is this change specific to entry point methods? Can we add tests for this?

Comment (lewing) [reply src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs:329]: It isn't

File: src/mono/wasm/runtime-test.js [hunk 2]
```diff
@@ -364,8 +364,7 @@ var App = {
 			}
 
 			try {
-				// Automatic signature isn't working correctly
-				let exit_code = Module.mono_call_assembly_entry_point (main_assembly_name, [app_args], "m");
+				let exit_code = Module.mono_call_assembly_entry_point (main_assembly_name, [app_args]);
```
Comment (lewing) [inline src/mono/wasm/runtime-test.js:367]: I think it is fine to keep the manual signature here.

**Review by lewing — COMMENTED** [review]
Is this really the right fix? What is special about string[]? It looks like the problem here is that the bindings assume the caller is passing a handle not a pointer to an object that was constructed manually.

**Comment by kjpou1** [comment]
Not needed

**Pull request closed without merging**